src/System.Device.Gpio/System/Device/Gpio/GpioController.Linux.cs

```diff
@@ -30,6 +31,17 @@ public GpioController(PinNumberingScheme numberingScheme)
         /// <returns>A driver that works with the board the program is executing on.</returns>
         private static GpioDriver GetBestDriverForBoard()
         {
+            // a first chance of finding a raspberry pi..... perhaps not on Raspbian note 
+            // that we want this to fail quietly if something goes wrong with the test
+            try
+            {
+                if (File.ReadAllText(ModelInfoPath).StartsWith("Raspberry Pi"))
```

Comment:
Why did you have to add this? was the current way of getting the right device not picking the Raspberry Pi Driver for the Pi 4? I would have expect the logic that we have after this with the Regex to have worked unless they changed their hardware which is very possible.

Reply:
It would only pick the raspberry pi for Raspbian. Debian does not include the chipset in /proc/cpuinfo (this may also be the case for Ubunto but I cannot remember right now). Reading the device tree allows detection of the pi regardless of Linux distribution. I did decide to leave the original detection code in as I didn't want to break something that was currently working but I was unaware of.

Reply:
I see, good to know. Actually it may very well be that your way of reading the device tree is better so we might want to switch our main logic to use this regardless of the /dev/mem work. That said, I'm curious if it would catch the case when running in an old RaspberryPi like the Pi1. That one I believe doesn't work with the RaspberryPi Driver so we needed to fallback to SysFs.

Reply:
Is the Pi 1 supported ? The docs for core seem to indicate that it isn't i.e.

> Note: All models of generation 1 and Pi Zero are not supported because the .NET Core JIT depends on armv7 instructions not available on those versions.

Perhaps I could find a Pi 2

Reply:
oh right that's a good point. I'll do some tests of this in our end as well, but I guess it's fine to keep it in your PR for now in order to get Debian support using this driver. Does Debian have /dev/gpiomem?

Reply:
No, Debian appears to just have /dev/mem. At the moment it seems very unfinished but was useful for testing Arm64 et. al.

Reply:
I have tested on a Pi 2 and it works as expected using /dev/gpiomem